lib/pbench/server/api/resources/query_apis/datasets_publish.py

```diff
@@ -188,25 +201,37 @@ def postprocess(self, es_json: JSON, context: JSON) -> JSON:
                     }
                 },
         """
-        msg = ""
-        status = HTTPStatus.OK
+        error_count = 0
         if es_json["errors"]:
             for i in es_json["items"]:
                 u = i["update"]
                 if u["status"] != HTTPStatus.OK:
                     e = u["error"]
-                    self.logger.error(
+                    self.logger.warning(
                         "{} ({}: {}) for id {} in index {}",
                         u["status"],
                         e["type"],
                         e["reason"],
                         u["_id"],
                         u["_index"],
                     )
-                    msg = "One or more Elasticsearch PUBLISH operations failed; see server log for details."
-                    status = HTTPStatus.INTERNAL_SERVER_ERROR
-        if status == HTTPStatus.OK:
-            dataset = context["dataset"]
+                    error_count += 1
+        if error_count > 0:
+            abort(
+                HTTPStatus.INTERNAL_SERVER_ERROR,
```

Comment:
My only suggestion is to relay one of the ES failures on to the user, in the hope that it will prove to be useful, actionable, or, at least, more mollifying than "it failed".